src/main/java/de/adesso/anki/battle/protocol/commands/FireRocketCommand.java

```diff
@@ -1,42 +1,39 @@
-package com.commands;
-
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.stereotype.Component;
-
-import de.adesso.anki.battle.world.World;
-import de.adesso.anki.battle.world.bodies.Rocket;
-import de.adesso.anki.battle.world.bodies.Vehicle;
-import lombok.extern.slf4j.Slf4j;
-
-
-
-@Slf4j
-public class FireRocketCommand extends Command {
-	private String direction;  // forward or backwards
-
-	
-	public FireRocketCommand(String direction) {
-		this.direction = direction;
-	}
-	
-	
-	public void execute(Vehicle vehicle) {
-		// TODO: set Positions
-		if (!vehicle.isRocketReady()) {
-			return;
-		}
-		World world = vehicle.getWorld();		
-		Rocket rocket  = new Rocket (this.direction);
-		//TODO speed of rocket
-		rocket.setTargetSpeed(300);
-		rocket.setPosition(vehicle.getPosition());
-		//rocket.setCurrentRoadpiece(vehicle.getCurrentRoadpiece());
-		rocket.setWorld(world);
-		world.addBody(rocket);
-		vehicle.setRocketReady(false);
-	}
-	
-}
-
-
-
+package de.adesso.anki.battle.protocol.commands;
+
+import de.adesso.anki.battle.world.World;
+import de.adesso.anki.battle.world.bodies.Rocket;
+import de.adesso.anki.battle.world.bodies.Vehicle;
+import lombok.extern.slf4j.Slf4j;
+
+
+
+@Slf4j
+public class FireRocketCommand extends Command {
+	private String direction;  // forward or backwards
+
+	
+	public FireRocketCommand(String direction) {
+		this.direction = direction;
+	}
+	
+	
+	@Override
+	public void execute(Vehicle vehicle) {
+		// TODO: set Positions
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Complete the task associated to this TODO comment. ([squid:S1135](https://sonarcloud.io/organizations/adessoag-github/rules#rule_key=squid%3AS1135))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=de.adesso.anki%3Aanki-battle&pullRequest=22&issues=AWKWIFfx1B6WA4TguGK8&open=AWKWIFfx1B6WA4TguGK8)

[ISSUE_KEY]: AWKWIFfx1B6WA4TguGK8